framework/src/main/java/org/checkerframework/framework/ajava/InsertAjavaAnnotations.java

```diff
@@ -0,0 +1,580 @@
+         * @param contents contents of the insertion
+         */
+        public Insertion(int position, String contents) {
+            this(position, contents, false);
+        }
+
+        /**
+         * Constructs an insertion with the given position and contents.
+         *
+         * @param position offset of the insertion in the file
+         * @param contents contents of the insertion
+         * @param ownLine true if this insertion represents an object on its own line (doesn't
+         *     affect the contents of the insertion)
+         */
+        public Insertion(int position, String contents, boolean ownLine) {
+            this.position = position;
+            this.contents = contents;
+            this.ownLine = ownLine;
+        }
+
+        @Override
+        public String toString() {
+            return "Insertion [contents=" + contents + ", position=" + position + "]";
+        }
+    }
+
+    /**
+     * Given two JavaParser ASTs representing the same Java file but with differing annotations,
+     * stores a list of {@link Insertion}s for all annotations in the first AST into the second AST.
+     */
+    private class BuildInsertionsVisitor extends DoubleJavaParserVisitor {
+        /**
+         * The set of annotations found in the file. Keys are both fully-qualified and simple names.
+         * There are two entries for each annotation: the annotation's simple name and its
+         * fully-qualified name.
+         *
+         * <p>The map is populated from import statements and also by when parsing a file that uses
+         * the fully qualified name of an annotation it doesn't import.
+         */
+        private Map<String, TypeElement> allAnnotations;
+
+        /** The annotation insertions seen so far. */
+        public List<Insertion> insertions;
+        /** A printer for annotations. */
+        private PrettyPrinter printer;
+        /** The lines of the String representation of the second AST. */
+        private List<String> lines;
+        /**
+         * Stores the offsets of the lines in the string representation of the second AST. At index
+         * i, stores the number of characters from the start of the file to the beginning of the ith
+         * line.
+         */
+        private List<Integer> cumulativeLineSizes;
+
+        /**
+         * Constructs a {@code BuildInsertionsVisitor} where {@code destFileContents} is the String
+         * representation of the AST to insertion annotations to. When visiting a node pair, the
+         * second node must always be from an AST generated from this String.
+         *
+         * @param destFileContents the String the second vistide AST was parsed from
+         */
+        public BuildInsertionsVisitor(String destFileContents) {
+            allAnnotations = null;
+            insertions = new ArrayList<>();
+            printer = new PrettyPrinter();
+            String[] lines = destFileContents.split(System.lineSeparator());
```

Comment:
This looks suspicious to me.  I occasionally process "foreign" files -- created under Windows or on a Mac -- and it would be best if the program handle those.  If you want to defer that to a later pull request, at least note the limitation.